cli/tools/coverage.rs

```diff
@@ -117,17 +113,250 @@ pub struct GetScriptSourceResult {
+pub struct LcovCoverageReporter {}
+
+impl LcovCoverageReporter {
+  pub fn new() -> LcovCoverageReporter {
+    LcovCoverageReporter {}
+  }
+}
+
+impl CoverageReporter for LcovCoverageReporter {
+  fn visit_coverage(
+    &mut self,
+    script_coverage: &ScriptCoverage,
+    script_source: &str,
+    maybe_source_map: Option<Vec<u8>>,
+    _maybe_original_source: Option<String>,
+  ) {
+    // TODO(caspervonb) cleanup and reduce duplication between reporters, pre-compute line coverage
+    // elsewhere.
+    let maybe_source_map = if let Some(source_map) = maybe_source_map {
+      Some(SourceMap::from_slice(&source_map).unwrap())
+    } else {
+      None
+    };
+
+    let url = Url::parse(&script_coverage.url).unwrap();
+    let file_path = url.to_file_path().unwrap();
+    println!("SF:{}", file_path.to_str().unwrap());
+
+    let mut functions_found = 0;
+    for function in &script_coverage.functions {
+      if function.function_name.is_empty() {
+        continue;
+      }
+
+      let source_line = script_source[0..function.ranges[0].start_offset]
+        .split('\n')
+        .count();
+
+      let line_index = if let Some(source_map) = maybe_source_map.as_ref() {
+        source_map
+          .tokens()
+          .find(|token| token.get_dst_line() as usize == source_line)
+          .map(|token| token.get_src_line() as usize)
+          .unwrap_or(0)
+      } else {
+        source_line
+      };
+
+      let function_name = &function.function_name;
+
+      println!("FN:{},{}", line_index + 1, function_name);
+
+      functions_found += 1;
+    }
+
+    let mut functions_hit = 0;
+    for function in &script_coverage.functions {
+      if function.function_name.is_empty() {
+        continue;
+      }
+
+      let execution_count = function.ranges[0].count;
+      let function_name = &function.function_name;
+
+      println!("FNDA:{},{}", execution_count, function_name);
+
+      if execution_count != 0 {
+        functions_hit += 1;
+      }
+    }
+
+    println!("FNF:{}", functions_found);
+    println!("FNH:{}", functions_hit);
+
+    let mut branches_found = 0;
+    let mut branches_hit = 0;
+    for (block_number, function) in script_coverage.functions.iter().enumerate()
+    {
+      for (branch_number, range) in function.ranges[1..].iter().enumerate() {
+        let source_line =
+          script_source[0..range.start_offset].split('\n').count();
+
+        let line_index = if let Some(source_map) = maybe_source_map.as_ref() {
+          source_map
+            .tokens()
+            .find(|token| token.get_dst_line() as usize == source_line)
+            .map(|token| token.get_src_line() as usize)
+            .unwrap_or(0)
+        } else {
+          source_line
+        };
+
+        let taken = range.count;
+
+        println!(
+          "BRDA:{},{},{},{}",
+          line_index + 1,
+          block_number,
+          branch_number,
+          taken
```

Comment:
According to https://manpages.debian.org/unstable/lcov/geninfo.1.en.html, taken should be `-` if the branches' surrounding block was not executed. Is this relevant for us?

Reply:
Huh, so seems v8 just doesn't emit branch info if the outer block was never hit.

Reply:
Seems we don't ever hit this state, left a comment and might have to come back here and do parse and traverse the source as the baseline for what "can be hit".